embrace-android-features/src/main/kotlin/io/embrace/android/embracesdk/internal/capture/aei/AeiDataSourceImpl.kt

```diff
@@ -43,193 +38,42 @@ internal class AeiDataSourceImpl(
         private const val SDK_AEI_SEND_LIMIT = 32
     }
 
-    @Volatile
-    private var backgroundExecution: Future<*>? = null
-    private val sessionApplicationExitInfoData: MutableList<AppExitInfoData> = mutableListOf()
-    private val isSessionApplicationExitInfoDataReady = AtomicBoolean(false)
-
     override fun enableDataCapture() {
-        if (backgroundExecution != null) {
-            return
-        }
-        backgroundExecution = backgroundWorker.submit {
+        backgroundWorker.submit {
             try {
-                processApplicationExitInfo()
+                processAeiRecords()
             } catch (exc: Throwable) {
                 logger.trackInternalError(InternalErrorType.ENABLE_DATA_CAPTURE, exc)
             }
         }
     }
 
-    override fun disableDataCapture() {
-        try {
-            backgroundExecution?.cancel(true)
-            backgroundExecution = null
-        } catch (t: Throwable) {
-            logger.trackInternalError(InternalErrorType.DISABLE_DATA_CAPTURE, t)
-        }
-    }
-
-    private fun processApplicationExitInfo() {
-        val historicalProcessExitReasons = getHistoricalProcessExitReasons()
-        val unsentExitReasons = getUnsentExitReasons(historicalProcessExitReasons)
-
-        unsentExitReasons.forEach {
-            sessionApplicationExitInfoData.add(buildSessionAppExitInfoData(it, null, null))
-        }
-
-        isSessionApplicationExitInfoDataReady.set(true)
-        processApplicationExitInfoBlobs(unsentExitReasons)
-    }
-
-    private fun processApplicationExitInfoBlobs(unsentExitReasons: List<ApplicationExitInfo>) {
-        unsentExitReasons.forEach { aei: ApplicationExitInfo ->
-            val traceResult = collectExitInfoTrace(aei)
-            if (traceResult != null) {
-                val payload = buildSessionAppExitInfoData(
-                    aei,
-                    getTrace(traceResult),
-                    getTraceStatus(traceResult)
-                )
-                sendApplicationExitInfoWithTraces(listOf(payload))
-            }
-        }
-    }
-
-    private fun getHistoricalProcessExitReasons(): List<ApplicationExitInfo> {
-        // A process ID that used to belong to this package but died later;
-        // a value of 0 means to ignore this parameter and return all matching records.
-        val pid = 0
-
-        // number of results to be returned; a value of 0 means to ignore this parameter and return
-        // all matching records with a maximum of 16 entries
+    private fun processAeiRecords() {
         val maxNum = appExitInfoBehavior.appExitInfoMaxNum()
```

Comment:
Given that we are dealing with all exists now rather than just those with blobs, is the limit we are setting still reasonable?

Reply:
That will be tackled when we implement the feature. I haven't made any functional changes as part of this refactor.